src/main/java/filter/expression/Qualifier.java

```diff
@@ -717,6 +714,61 @@ private boolean milestoneSatisfies(IModel model, TurboIssue issue) {
         String contents = content.get().toLowerCase();
         String title = milestone.get().getTitle().toLowerCase();
 
+        List<TurboMilestone> allMilestones = model.getMilestones();
+        Collections.sort(allMilestones, new TurboMilestoneDueDateComparator());
+
+        int currentIndex = -1;
+
+        for (TurboMilestone checker : allMilestones) {
+            // current milestone must be open
+            if (checker.isOpen()) {
+                // if milestone is overdue but have no open issue, ignore
+                if (checker.getDueDate().isPresent() &&
+                        checker.getDueDate().get().isBefore(LocalDate.now()) &&
+                        checker.getOpenIssues() == 0) {
+                    continue;
+                }
+                currentIndex = allMilestones.indexOf(checker);
+                break;
+            }
+        }
+
+        // if no open milestone, set current as last milestone
+        if (currentIndex == -1 && allMilestones.size() > 0) {
+            currentIndex = allMilestones.size() - 1;
+        }
+
+        String alias = contents;
+
+        if (alias.equalsIgnoreCase("current") ||
+                alias.equalsIgnoreCase("curr")) {
+            contents = allMilestones.get(currentIndex).getTitle();
+        } else if (alias.equalsIgnoreCase("previous") ||
+                alias.equalsIgnoreCase("prev")) {
+            if (currentIndex - 1 >= 0) {
+                contents = allMilestones.get(currentIndex - 1).getTitle();
+            }
+        } else if (alias.equalsIgnoreCase("next")) {
+            if (currentIndex + 1 < allMilestones.size()) {
+                contents = allMilestones.get(currentIndex + 1).getTitle();
+            }
+        } else if (alias.startsWith("previous-") ||
+                alias.startsWith("prev-")) {
+            int offsetPos = alias.indexOf('-') + 1;
+            int offset = Integer.parseInt(alias.substring(offsetPos).trim());
+            if (currentIndex - 1 - offset >= 0) {
+                contents = allMilestones.get(currentIndex - 1 - offset).getTitle();
+            }
+        } else if (alias.startsWith("next+")) {
+            int offsetPos = alias.indexOf('+') + 1;
+            int offset = Integer.parseInt(alias.substring(offsetPos).trim());
+            if (currentIndex + 1 + offset < allMilestones.size()) {
+                contents = allMilestones.get(currentIndex + 1 + offset).getTitle();
+            }
+        }
+
+        contents = contents.toLowerCase();
+
```

Comment:
But even if I make regex, it would still have that many `if` cases. And since each alias has its own kind of 'signature', the regex would just be as many, wouldn't it?

I noticed that this method will be invoked for every issue, but where should preprocessing of the milestones be done? In `Model`/`MultiModel`? That's the only place I can think of since putting it in `Qualifier` would not make it invoked once. In that case, should the milestone comparator be extracted into its own class so that it can be used by sort milestone as well?

On a side note, I might need to change parts of this since right now, it will refer to milestones of all open repos; so if there is a panel using `repo:sth/sth`, that repo's milestones will be listed together instead of only the relevant repo to the filter. I am not sure how to do it yet, though.